<!DOCTYPE html>
<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8"/>

<title>Trying to test which measure feels better that chrY methylation is represented in female samples less than in male ones</title>

<script type="text/javascript">
window.onload = function() {
  var imgs = document.getElementsByTagName('img'), i, img;
  for (i = 0; i < imgs.length; i++) {
    img = imgs[i];
    // center an image if it is the only element of its parent
    if (img.parentElement.childElementCount === 1)
      img.parentElement.style.textAlign = 'center';
  }
};
</script>





<style type="text/css">
body, td {
   font-family: sans-serif;
   background-color: white;
   font-size: 13px;
}

body {
  max-width: 800px;
  margin: auto;
  padding: 1em;
  line-height: 20px;
}

tt, code, pre {
   font-family: 'DejaVu Sans Mono', 'Droid Sans Mono', 'Lucida Console', Consolas, Monaco, monospace;
}

h1 {
   font-size:2.2em;
}

h2 {
   font-size:1.8em;
}

h3 {
   font-size:1.4em;
}

h4 {
   font-size:1.0em;
}

h5 {
   font-size:0.9em;
}

h6 {
   font-size:0.8em;
}

a:visited {
   color: rgb(50%, 0%, 50%);
}

pre, img {
  max-width: 100%;
}
pre {
  overflow-x: auto;
}
pre code {
   display: block; padding: 0.5em;
}

code {
  font-size: 92%;
  border: 1px solid #ccc;
}

code[class] {
  background-color: #F8F8F8;
}

table, td, th {
  border: none;
}

blockquote {
   color:#666666;
   margin:0;
   padding-left: 1em;
   border-left: 0.5em #EEE solid;
}

hr {
   height: 0px;
   border-bottom: none;
   border-top-width: thin;
   border-top-style: dotted;
   border-top-color: #999999;
}

@media print {
   * {
      background: transparent !important;
      color: black !important;
      filter:none !important;
      -ms-filter: none !important;
   }

   body {
      font-size:12pt;
      max-width:100%;
   }

   a, a:visited {
      text-decoration: underline;
   }

   hr {
      visibility: hidden;
      page-break-before: always;
   }

   pre, blockquote {
      padding-right: 1em;
      page-break-inside: avoid;
   }

   tr, img {
      page-break-inside: avoid;
   }

   img {
      max-width: 100% !important;
   }

   @page :left {
      margin: 15mm 20mm 15mm 10mm;
   }

   @page :right {
      margin: 15mm 10mm 15mm 20mm;
   }

   p, h2, h3 {
      orphans: 3; widows: 3;
   }

   h2, h3 {
      page-break-after: avoid;
   }
}
</style>



</head>

<body>
<h1>Trying to test which measure feels better that chrY methylation is represented in female samples less than in male ones</h1>

<h2>Alexander Favorov </h2>

<h3>09 Nov 2014</h3>

<p>We will test three measures: the coverage of methylation MACS peaks, their score (summed over chromosome) and their product.
To do this, we use an absurd comparison of methylation of Y-chromosome between males and females.</p>

<p>We will make Wilcoxon test for the Y-chr methylation mesure levels and we will make a test of 1&#39;s for all males against 0&#39;s for all females to have a watermark (the last comparison).</p>

<p>Here is coverage of Y-chromome of all the samples by the methylated peaks, the sum of all the scores and the product of length and scores for all the samples.</p>

<pre><code>##    code        DNAid tumor  male Peak.length.sum Peak.score.sum Product.sum
## 1     1        37920     1  TRUE          414190      105236.59   129432813
## 2     2        37941     1  TRUE          413742      101598.35   207614610
## 3     3        41791     1  TRUE          446762      116579.89   147218443
## 4     4        39451     1  TRUE          527631      151385.26   190243317
## 5     5        34346     1  TRUE          546384      154164.82   206414713
## 6     6        35667     1 FALSE          376294      102885.42   150700149
## 7     7        27254     1 FALSE          505291      135354.40   184599984
## 8     8         3366     1  TRUE          486388      121827.22   157717673
## 9     9    3680_TDNA     1  TRUE          509937      140122.85   159298330
## 10   10        35675     1 FALSE          413638      104125.63   147113371
## 11   11        41505     1  TRUE          232460       45588.17    48404062
## 12   12    2934_TDNA     1  TRUE          547344      169367.75   197774550
## 13   13        24366     1  TRUE          448006      115735.68   112974761
## 14   14         4931     1  TRUE          651050      162214.79   220521307
## 15   15        32085     1  TRUE          421773      112761.96   125925824
## 16   16        18732     1  TRUE          564458      145679.20   230736378
## 17   17        40666     1  TRUE          367988       99783.05   133307083
## 18   18        31124     1  TRUE          403700       91285.04   121400575
## 19   19        35663     1  TRUE          413478      103125.46   132697472
## 20   20        41522     1  TRUE          236881       49224.00    50169334
## 21   21        23896     1  TRUE          403758       80055.48    86827917
## 22   22        38667     1  TRUE          533276      113744.17   111625263
## 23   23        38623     1  TRUE          310553       87293.50   118577144
## 24   24        35639     1  TRUE          794539      219144.24   283511321
## 25   25        27281     1  TRUE          467948      128386.07   192180775
## 26   26        39713     1  TRUE          376161      105557.89   140902401
## 27   27 43281, 28680     1  TRUE          557718      125008.32   142257086
## 28   28        40908     1  TRUE          380306       88631.40   111908676
## 29   29        27279     1  TRUE          430185       97185.36   107257870
## 30   30        40913     1  TRUE          359289       96963.65   140588958
## 31   31        23749     1  TRUE          542038      134365.17   161772583
## 32   32        38676     1  TRUE          534061      167044.75   203484652
## 33   33        27239     1  TRUE          535922      156848.00   176504492
## 34   34        38851     1  TRUE          364894       80495.72   111868956
## 35   35        37960     1 FALSE          317636       86824.22   102754483
## 36   36        30648     1  TRUE          374560       88669.40   139099195
## 37   37        41790     1  TRUE          373763       63775.02    86650347
## 38   38    3669_TDNA     1  TRUE          436423      136702.70   177671564
## 39   39        23947     1  TRUE          415222      107023.85   134675291
## 40   40         5006     1  TRUE          609799      186406.21   299441207
## 41   41  38690, 4842     1 FALSE          262013       68187.28    94879741
## 42   42        38698     1  TRUE          454850      126659.57   159065795
## 43   43        24866     1  TRUE          290144       64325.69    70576459
## 44   44         1342     1  TRUE          349875       84652.16   129851383
## 45   45        39477     1  TRUE          446071      137967.25   207801657
## 46   46        40673     1  TRUE          451897       93244.88   133826776
## 47   47    2704_TDNA     1  TRUE          423701      100965.69   147144639
## 48   48    2675_TDNA     1  TRUE          365662       86044.89   113260940
## 49   49    3714_TDNA     1  TRUE          449862      109639.95   136443581
## 50   50        40907     1  TRUE          490619      127413.35   179266955
## 51  101        25154     0 FALSE          419499       92613.84   154185703
## 52  102 43250, 25110     0 FALSE          371873       76455.99    77714846
## 53  103        25320     0 FALSE          274957       73797.89   112944666
## 54  104        25304     0 FALSE          246930       55096.92    93494064
## 55  105 43236, 25317     0 FALSE          459558      113123.56   138904015
## 56  106        25116     0  TRUE          495873      164327.22   200367330
## 57  107        34432     0 FALSE          464563      130189.26   164965603
## 58  108        27773     0  TRUE          537632      158909.68   185754957
## 59  109        27743     0  TRUE          483551      147446.23   175526118
## 60  110        34440     0 FALSE          535880      152499.58   158366699
## 61  111        34471     0  TRUE          409733       90571.10   117914505
## 62  112        34448     0 FALSE          387877      106682.80   119708536
## 63  113        30588     0 FALSE          325036       84087.09   126074278
## 64  114        30593     0 FALSE          275500       68870.23    95235278
## 65  115        30551     0 FALSE          306051       78078.06   135124306
## 66  116        30555     0 FALSE          339737       86748.81   115337876
## 67  117        30599     0  TRUE          427455      108946.68   128314866
## 68  118        30539     0 FALSE          269001       63927.48    98069357
## 69  119        30553     0  TRUE          304730       62311.50    98240747
## 70  120        30596     0  TRUE          763616      207376.46   366795930
## 71  121        30577     0  TRUE          462897      114263.73   164465789
## 72  122        30580     0  TRUE          518039      127543.92   187512987
## 73  123        30018     0 FALSE          239088       35748.24    32169528
## 74  124        31415     0  TRUE          359808       95687.55   148503276
## 75  125        32913     0 FALSE          392266       93406.98   155932321
</code></pre>

<p>Wilcoxon test of the methylation peaks coverage the Y-chromosome for males/females </p>

<pre><code>## 
##  Wilcoxon rank sum test with continuity correction
## 
## data:  Y.chrom.stats$Peak.length.sum[Y.chrom.stats$male] and Y.chrom.stats$Peak.length.sum[!Y.chrom.stats$male]
## W = 823, p-value = 0.001095
## alternative hypothesis: true location shift is not equal to 0
</code></pre>

<p>Wilcoxon test of the scores sum of Y-chromosome methylation peaks for males/females </p>

<pre><code>## 
##  Wilcoxon rank sum test with continuity correction
## 
## data:  Y.chrom.stats$Peak.score.sum[Y.chrom.stats$male] and Y.chrom.stats$Peak.score.sum[!Y.chrom.stats$male]
## W = 798, p-value = 0.003024
## alternative hypothesis: true location shift is not equal to 0
</code></pre>

<p>Wilcoxon test of the scores*length product sum Y-chromosome for males/females </p>

<pre><code>## 
##  Wilcoxon rank sum test with continuity correction
## 
## data:  Y.chrom.stats$Product.sum[Y.chrom.stats$male] and Y.chrom.stats$Product.sum[!Y.chrom.stats$male]
## W = 743, p-value = 0.02109
## alternative hypothesis: true location shift is not equal to 0
</code></pre>

<p>The watermark test:  Wilcoxon for 1&#39;s for all males against 0&#39;s for all females.</p>

<pre><code>## 
##  Wilcoxon rank sum test with continuity correction
## 
## data:  one_m_j and zero_f_j
## W = 1100, p-value = 4.596e-11
## alternative hypothesis: true location shift is not equal to 0
</code></pre>

<h3>The best measure is the coverage lenghts.</h3>

<h2>Technical info</h2>

<p>Clinical data file: ../../Data/Clinical/2014-10-07_DiscoveryCohort_ClinicalData-Final_DeID.csv.</p>

<p>Its size and modification time: </p>

<pre><code>##                                                                             size
## ../../Data/Clinical/2014-10-07_DiscoveryCohort_ClinicalData-Final_DeID.csv 14084
##                                                                                          mtime
## ../../Data/Clinical/2014-10-07_DiscoveryCohort_ClinicalData-Final_DeID.csv 2014-10-27 11:54:21
</code></pre>

<p>Number of records is 79</p>

<p>Number of used records is 75</p>

<p>There were 75 bed files:</p>

<pre><code>##  [1] &quot;002_38623_Enrich_hg19.combined_vs_001_38623_Total_hg19.combined.50_peaks.bed&quot;                       
##  [2] &quot;004_38676_Enrich_hg19.combined_vs_003_38676_Total_hg19.combined.50_peaks.bed&quot;                       
##  [3] &quot;004_41505_enrich_hg19.combined_vs_003_41505_total_hg19.combined.50_peaks.bed&quot;                       
##  [4] &quot;005_31415_Enrich_hg19.combined_vs_016_31415_Total_hg19.combined.50_peaks.bed&quot;                       
##  [5] &quot;006_34440_Enrich_hg19.combined_vs_017_34440_Total_hg19.combined.50_peaks.bed&quot;                       
##  [6] &quot;006_39713_enrich_hg19.combined_vs_005_39713_total_hg19.combined.50_peaks.bed&quot;                       
##  [7] &quot;008_27743_Enrich_hg19.combined_vs_014_27743_Total_hg19.combined.50_peaks.bed&quot;                       
##  [8] &quot;008_41791_enrich_hg19.combined_vs_007_41791_total_hg19.combined.50_peaks.bed&quot;                       
##  [9] &quot;009_39477_enrich_hg19.combined_vs_001_39477_total_hg19.combined.50_peaks.bed&quot;                       
## [10] &quot;009_40913_enrich_hg19.combined_vs_001_40913_total_hg19.combined.50_peaks.bed&quot;                       
## [11] &quot;010_34432_enrich_hg19.combined_vs_003_34432_total_hg19.combined.50_peaks.bed&quot;                       
## [12] &quot;010_SCC_090_enrich_hg19.combined_vs_002_SCC_090_total_hg19.combined.50_peaks.bed&quot;                   
## [13] &quot;011_25320_enrich_hg19.combined_vs_003_25320_total_hg19.combined.50_peaks.bed&quot;                       
## [14] &quot;011_3669_TDNA_enrich_hg19.combined_vs_002_3669_TDNA_total_hg19.combined.50_peaks.bed&quot;               
## [15] &quot;012_39451_enrich_hg19.combined_vs_004_39451_total_hg19.combined.50_peaks.bed&quot;                       
## [16] &quot;012_40673_enrich_hg19.combined_vs_004_40673_total_hg19.combined.50_peaks.bed&quot;                       
## [17] &quot;012_41522_enrich_hg19.combined_vs_011_41522_total_hg19.combined.50_peaks.bed&quot;                       
## [18] &quot;013_25116_enrich_hg19.combined_vs_005_25116_total_hg19.combined.50_peaks.bed&quot;                       
## [19] &quot;013_3714_TDNA_enrich_hg19.combined_vs_005_3714_TDNA_total_hg19.combined.50_peaks.bed&quot;               
## [20] &quot;013_42133_enrich_hg19.combined_vs_014_42133_total_hg19.combined.50_peaks.bed&quot;                       
## [21] &quot;014_25154_enrich_hg19.combined_vs_006_25154_total_hg19.combined.50_peaks.bed&quot;                       
## [22] &quot;014_43684_rpt2_enrich_hg19.combined_vs_013_43684_rpt2_total_hg19.combined.50_peaks.bed&quot;             
## [23] &quot;014_5006_enrich_hg19.combined_vs_006_5006_total_hg19.combined.50_peaks.bed&quot;                         
## [24] &quot;015_27254_Enrich_hg19.combined_vs_013_27254_Total_hg19.combined.50_peaks.bed&quot;                       
## [25] &quot;015_27773_enrich_hg19.combined_vs_007_27773_total_hg19.combined.50_peaks.bed&quot;                       
## [26] &quot;015_40907_enrich_hg19.combined_vs_007_40907_total_hg19.combined.50_peaks.bed&quot;                       
## [27] &quot;016_2704_TDNA_enrich_hg19.combined_vs_008_2704_TDNA_total_hg19.combined.50_peaks.bed&quot;               
## [28] &quot;016_34346_enrich_hg19.combined_vs_008_34346_total_hg19.combined.50_peaks.bed&quot;                       
## [29] &quot;016_40908_enrich_hg19.combined_vs_015_40908_total_hg19.combined.50_peaks.bed&quot;                       
## [30] &quot;016_41790_rpt2_enrich_hg19.combined_vs_015_41790_rpt2_total_hg19.combined.50_peaks.bed&quot;             
## [31] &quot;018_SCC_047_2_rpt2_enrich_hg19.combined_vs_017_SCC_047_2_rpt2_total_hg19.combined.50_peaks.bed&quot;     
## [32] &quot;020_43281_rpt2_enrich_hg19.combined_vs_019_43281_rpt2_total_hg19.combined.50_peaks.bed&quot;             
## [33] &quot;021_37960_Enrich_hg19.combined_vs_007_37960_Total_hg19.combined.50_peaks.bed&quot;                       
## [34] &quot;022_43250_rpt2_enrich_hg19.combined_vs_021_43250_rpt2_total_hg19.combined.50_peaks.bed&quot;             
## [35] &quot;024_38698_Enrich_hg19.combined_vs_023_38698_Total_hg19.combined.50_peaks.bed&quot;                       
## [36] &quot;024_43236_rpt2_enrich_hg19.combined_vs_023_43236_rpt2_total_hg19.combined.50_peaks.bed&quot;             
## [37] &quot;025_3366_Enrich_hg19.combined_vs_009_3366_Total_hg19.combined.50_peaks.bed&quot;                         
## [38] &quot;025_38690_enrich_hg19.combined_vs_017_38690_total_hg19.combined.50_peaks.bed&quot;                       
## [39] &quot;026_23749_Enrich_hg19.combined_vs_010_23749_Total_hg19.combined.50_peaks.bed&quot;                       
## [40] &quot;027_23896_Enrich_hg19.combined_vs_011_23896_Total_hg19.combined.50_peaks.bed&quot;                       
## [41] &quot;027_40666_enrich_hg19.combined_vs_019_40666_total_hg19.combined.50_peaks.bed&quot;                       
## [42] &quot;028_23947_Enrich_hg19.combined_vs_012_23947_Total_hg19.combined.50_peaks.bed&quot;                       
## [43] &quot;028_25304_enrich_hg19.combined_vs_020_25304_total_hg19.combined.50_peaks.bed&quot;                       
## [44] &quot;029_34448_Enrich_hg19_vs_018_34448_Total_hg19.combined.50_peaks.bed&quot;                                
## [45] &quot;029_38851_enrich_hg19.combined_vs_021_38851_total_hg19.combined.50_peaks.bed&quot;                       
## [46] &quot;030_35639_Enrich_hg19.combined_vs_020_35639_Total_hg19.combined.50_peaks.bed&quot;                       
## [47] &quot;031_38667_Enrich_hg19.combined_vs_022_38667_Total_hg19.combined.50_peaks.bed&quot;                       
## [48] &quot;032_34471_Enrich_hg19.combined_vs_019_34471_Total_hg19.combined.50_peaks.bed&quot;                       
## [49] &quot;18732_enrich.combined.removeDuplicates_vs_18732_total.combined.removeDuplicates.50_peaks.bed&quot;       
## [50] &quot;25441_enrich.combined.removeDuplicates_vs_25441_total.combined.removeDuplicates.50_peaks.bed&quot;       
## [51] &quot;27279_enrich.combined.removeDuplicates_vs_27279_total.combined.removeDuplicates.50_peaks.bed&quot;       
## [52] &quot;27281_enrich.combined.removeDuplicates_vs_27281_total.combined.removeDuplicates.50_peaks.bed&quot;       
## [53] &quot;30018_enrich.combined.removeDuplicates_vs_30018_total.combined.removeDuplicates.50_peaks.bed&quot;       
## [54] &quot;30553_enrich.combined.removeDuplicates_vs_30553_total.combined.removeDuplicates.50_peaks.bed&quot;       
## [55] &quot;30577_enrich.combined.removeDuplicates_vs_30577_total.combined.removeDuplicates.50_peaks.bed&quot;       
## [56] &quot;30580_enrich.combined.removeDuplicates_vs_30580_total.combined.removeDuplicates.50_peaks.bed&quot;       
## [57] &quot;30596_enrich.combined.removeDuplicates_vs_30596_total.combined.removeDuplicates.50_peaks.bed&quot;       
## [58] &quot;31124_enrich.combined.removeDuplicates_vs_31124_total.combined.removeDuplicates.50_peaks.bed&quot;       
## [59] &quot;32913_enrich.combined.removeDuplicates_vs_32913_total.combined.removeDuplicates.50_peaks.bed&quot;       
## [60] &quot;35663_enrich.combined.removeDuplicates_vs_35663_total.combined.removeDuplicates.50_peaks.bed&quot;       
## [61] &quot;35667_enrich.combined.removeDuplicates_vs_35667_total.combined.removeDuplicates.50_peaks.bed&quot;       
## [62] &quot;35675_enrich.combined.removeDuplicates_vs_35675_total.combined.removeDuplicates.50_peaks.bed&quot;       
## [63] &quot;37941_enrich.combined.removeDuplicates_vs_37941_total.combined.removeDuplicates.50_peaks.bed&quot;       
## [64] &quot;4931_enrich.combined.removeDuplicates_vs_4931_total.combined.removeDuplicates.50_peaks.bed&quot;         
## [65] &quot;JHUJC01002_001_37920_enrich_hg19.combined_vs_JHUJC01002_002_37920_total.combined.50_peaks.bed&quot;      
## [66] &quot;JHUJC01002_003_3680TDNA_enrich_hg19.combined_vs_JHUJC01002_004_3680TDNA_total.combined.50_peaks.bed&quot;
## [67] &quot;JHUJC01002_005_24366_enrich.combined_vs_JHUJC01002_006_24366_total.combined.50_peaks.bed&quot;           
## [68] &quot;JHUJC01002_007_2934TDNA_enrich.combined_vs_JHUJC01002_008_2934TDNA_total.combined.50_peaks.bed&quot;     
## [69] &quot;JHUJC01002_009_32085_enrich.combined_vs_JHUJC01002_010_32085_total.combined.50_peaks.bed&quot;           
## [70] &quot;JHUJC01002_011_27239_enrich.combined_vs_JHUJC01002_012_27239_total.combined.50_peaks.bed&quot;           
## [71] &quot;JHUJC01002_013_24866_enrich.combined_vs_JHUJC01002_014_24866_total.combined.50_peaks.bed&quot;           
## [72] &quot;JHUJC01002_015_30599_enrich.combined_vs_JHUJC01002_016_30599_total.combined.50_peaks.bed&quot;           
## [73] &quot;JHUJC01002_017_30593_enrich.combined.bam.JHUJC01002_018_30593_total.combined.bam_peaks.bed&quot;         
## [74] &quot;JHUJC01002_019_1342_enrich.combined.bam.JHUJC01002_020_1342_total.combined.bam_peaks.bed&quot;           
## [75] &quot;JHUJC01002_021_30588_enrich.combined.bam.JHUJC01002_022_30588_total.combined.bam_peaks.bed&quot;         
## [76] &quot;JHUJC01002_023_30551_enrich.combined.bam.JHUJC01002_024_30551_total.combined.bam_peaks.bed&quot;         
## [77] &quot;JHUJC01002_025_30555_enrich.combined.bam.JHUJC01002_026_30555_total.combined.bam_peaks.bed&quot;         
## [78] &quot;JHUJC01002_027_30539_enrich.combined.bam.JHUJC01002_028_30539_total.combined.bam_peaks.bed&quot;         
## [79] &quot;JHUJC01002_029_30648_enrich.combined.bam.JHUJC01002_030_30648_total.combined.bam_peaks.bed&quot;         
## [80] &quot;JHUJC01002_031_2675TDNA_enrich.combined.bam.JHUJC01002_032_2675TDNA_total.combined.bam_peaks.bed&quot;
</code></pre>

<p>Beds not used:</p>

<pre><code>## [1] &quot;010_SCC_090_enrich_hg19.combined_vs_002_SCC_090_total_hg19.combined.50_peaks.bed&quot;              
## [2] &quot;013_42133_enrich_hg19.combined_vs_014_42133_total_hg19.combined.50_peaks.bed&quot;                  
## [3] &quot;014_43684_rpt2_enrich_hg19.combined_vs_013_43684_rpt2_total_hg19.combined.50_peaks.bed&quot;        
## [4] &quot;018_SCC_047_2_rpt2_enrich_hg19.combined_vs_017_SCC_047_2_rpt2_total_hg19.combined.50_peaks.bed&quot;
## [5] &quot;25441_enrich.combined.removeDuplicates_vs_25441_total.combined.removeDuplicates.50_peaks.bed&quot;
</code></pre>

</body>

</html>
